WordPress/gutenberg
# Enforce @since tags in block-library/src/*/*.php files

## What?
This PR introduces a new sniff called `FunctionCommentSinceTagSniff` to ensure that PHP functions have a valid `@since` tag in the docblock.

The sniff skips checking files in `__experimental` block-library packages.

Fixes https://github.com/WordPress/gutenberg/issues/55777.

## Why?
`@since` tags are helpful for tracking origins and how long things have been in Core.

## How?
- [x] Defines a new sniff class `FunctionCommentSinceTagSniff` within the `GutenbergCS\Gutenberg\Sniffs\Commenting` namespace.
- [x] Adds unit tests for the `FunctionCommentSinceTagSniff` sniff.
- [x] Updates existing `block-library` functions so that they have valid `@since` tags.

## Testing Instructions
1. Try to define a PHP function somewhere in the `block-library/src/*/*.php` without a `@since` tag.
2. Make sure that CI checks fail.

### Testing Instructions for Keyboard


## Screenshots or Screencast
